pilot/cmd/pilot-agent/status/server.go

```diff
@@ -709,10 +709,13 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 	}
 
 	// Forward incoming headers to the application.
-	appReq.Host = req.Host
+	if prober.HTTPGet.Host != "" {
+		appReq.Host = prober.HTTPGet.Host
+	}
 	for name, values := range req.Header {
 		appReq.Header[name] = slices.Clone(values)
 		if len(values) > 0 && (strings.EqualFold(name, "Host") || name == ":authority") {
+			// Probe has specific host header override; honor it
```

Comment:
One thing I am not clear on...

isn't `req.Host == req.Header["Host"]`? So how can we have this code and its fine (I tested it -- it works great as is in the PR), but it still works.

Reply:
Not sure I totally understand your question here

Reply:
From Godoc:

```
	// For incoming requests, the Host header is promoted to the
	// Request.Host field and removed from the Header map.
```

So is it actually possible for this line to be hit?

Reply:
Maybe not for "Host" header but for the ":authority" header?

Reply:
I would think the same applies to both of those.. maybe not though